tests/e2e/koperator_test.go

```diff
@@ -0,0 +1,146 @@
+					SetValues: map[string]string{
+						"crd.enabled": "true",
+					},
+				},
+				"../../charts/kafka-operator",
+				"dummy",
+				[]string{"templates/crds.yaml"},
+			))
+		default:
+			httpClient := new(http.Client)
+			httpClient.Timeout = 5 * time.Second
+
+			response, err := httpClient.Get("https://github.com/banzaicloud/koperator/releases/download/" + koperatorVersion + "/kafka-operator.crds.yaml")
+			if response != nil {
+				defer func() { _ = response.Body.Close() }()
+			}
+
+			Expect(err).NotTo(HaveOccurred())
+
+			rawKoperatorCRDsManifest, err = io.ReadAll(response.Body)
+
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		rawKoperatorCRDs := bytes.Split(rawKoperatorCRDsManifest, []byte("\n---\n"))
+		for rawKoperatorCRDIndex, rawKoperatorCRD := range rawKoperatorCRDs {
+			rawKoperatorCRDs[rawKoperatorCRDIndex] = bytes.Trim(rawKoperatorCRD, "\n-")
+		}
+
+		allowedCRDByteCount := 262144
+		tempDir := os.TempDir()
+		tempFile := path.Join(tempDir, "manifest.yaml")
+		for _, rawKoperatorCRD := range rawKoperatorCRDs {
+			err = os.WriteFile(tempFile, rawKoperatorCRD, 0o777)
+
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(rawKoperatorCRD) > allowedCRDByteCount { // Note: too long CRDs cannot be applied, only created or replaced.
+				var koperatorCRD apiextensionsv1.CustomResourceDefinition
+				err := yaml.Unmarshal(rawKoperatorCRD, &koperatorCRD)
+
+				Expect(err).NotTo(HaveOccurred())
+
+				createOrReplaceK8sResourcesFromManifest(kubectlOptions, "crd", koperatorCRD.GetName(), tempFile, false)
+			} else {
+				By("Applying koperator CRDs with version " + koperatorVersion)
+				applyK8sResourceManifest(kubectlOptions, tempFile)
+			}
+		}
+
+		By("Verifying koperator CRDs")
+		requireExistingCRDs(
+			kubectlOptions,
+			"cruisecontroloperations.kafka.banzaicloud.io",
+			"kafkaclusters.kafka.banzaicloud.io",
+			"kafkatopics.kafka.banzaicloud.io",
+			"kafkausers.kafka.banzaicloud.io",
+		)
+	})
+}
+
+// requireInstallingKoperator deploys koperator CRDs and Helm chart and checks
+// the success of those operations.
+func requireInstallingKoperator(kubectlOptions *k8s.KubectlOptions, koperatorVersion Version) {
+	When("Installing koperator", func() {
+		requireApplyingKoperatorCRDs(kubectlOptions, koperatorVersion)
+		requireInstallingKoperatorHelmChartIfDoesNotExist(kubectlOptions, koperatorVersion)
+	})
+}
+
+// requireDeployingKoperatorHelmChart checks the existence of the koperator Helm
+// release and installs it if it's not present.
+func requireInstallingKoperatorHelmChartIfDoesNotExist(
+	kubectlOptions *k8s.KubectlOptions,
+	koperatorVersion Version,
+) {
+	It("Installing koperator Helm chart", func() {
+		switch koperatorVersion {
+		case LocalVersion:
+			installHelmChartIfDoesNotExist(kubectlOptions, "", "../../charts/kafka-operator", "", "kafka-operator", nil)
+		default:
+			installHelmChartIfDoesNotExist(
+				kubectlOptions,
+				"https://kubernetes-charts.banzaicloud.com",
+				"kafka-operator",
```

Comment:
I think we should move out the namespaces for the different components into global variable

Reply:
Fixed in b4f5d00.